README.md

```diff
@@ -122,6 +122,172 @@ your container. This will allow you to login to the root account remotely. Local
+```
+FROM <this-image>
+ADD mystuff /usr/bin/
+RUN /usr/bin/mystuff
+```
+
+### Example of adding a new configuration file in layered Dockerfile:
+
+To change configuration for the `mysqld` daemon and other utilities running within the container, place your new configuration files with `*.cnf` extention into `/etc/my.cnf.d` directory.
+
+Example of the layered Dockerfile:
+
+```
+FROM <this-image>
+ADD myconfig.cnf /etc/my.cnf.d/myconfig.cnf
+```
+
+Notice: Read order of configuration files in `/etc/my.cnf.d/` directory is not specified.
+
+### Adjusting database inicialization
+
+Initialization of mysql database during first start of this image is devided into several files in several directories. All directories include files that are read in alphabetical order, and thus file names are usually prefixed with a number.
+
+Concretly there are these directories within the Docker container:
+
+* `/usr/share/container-layer/mysql/pre-init/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before every inicialization of container (no matter whether the data directory is empty or not). At the time these scripts are executed, the daemon is not yet running. Example of script in this directory is a script that checks arguments specified by user.
+* `/usr/share/container-layer/mysql/post-init/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before the first start of container (i.e. in case the container is started with empty data directory). At the time these scripts are executed, the daemon runs on localhost only and `mysql` utility should use arguments stored in `$mysql_flags` environment variable. This directory usually includes scripts for creating new databases, users or initialize the database according to your needs.
+* `/usr/share/container-layer/mysql/pre-start/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before every start of container (no matter whether the data directory is empty or not). At the time these scripts are executed, the daemon runs on localhost only and `mysql` utility should use arguments stored in `$mysql_flags` environment variable. Example of script in this directory is a script that allows to change password.
+* `/usr/share/container-layer/mysql/include/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed in the beginning of the commands like run-mysqld. Files in this directory contain usually functions or variables definition that are supposed to be used in other scripts used in the image.
+* `/usr/share/container-layer/mysql/usage/` -- this directory includes text files with extention `.txt` that will be printed as usage message for the container in case wrong arguments are given. Shell variables in these files will be expanded using `envsubst` utility.
```

Comment:
Well, changing original file always implies that you must pay attention what you need to change and reflect the non-changed information in your copy. In case you change everything, you're way might be easier.. But I more counted with scenery, where you don't want to change everything, but just add something to existing feature set.